src/GeneralStateTestsFiller/stEWASMTests/README.md

```diff
@@ -0,0 +1,318 @@
+    value: !!int -1
+  network:
+    - ALL
+  result:
+    # InitialBalance - contract1.call - valueTransfer - txCost - contract2.calldatasize - contract2.sstore
+    # 100000000000   - 700            - 9000          - 21000  - 2                      - 20000            = 99999949298
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '99999949298'
+    deadbeef00000000000000000000000000000000:
+      storage: {
+      }
+    deadbeef00000000000000000000000000000001:
+      storage: {
+        0: "0x4"
+      }
+      balance: "100000000001"
+```
+
+For the expect section values of indexes and network remain the same in all test
+cases, the most important field in this section is `result`.
+
+In `result` field we specify the accounts as we do in the `pre` section, but now
+we define the values that we expect as result of the transaction and the
+execution of the wast code.
+
+```yaml
+deadbeef00000000000000000000000000000001:
+   storage: {
+     0: "0x4"
+   }
+   balance: "100000000001"
+```
+### 4. transaction
+
+Transaction contains the following fields:
+
+- data
+- gasLimit
+- gasPrice - usually set to 1 to make it easy to calculate gas
+- nonce
+- secretKey - using
+  `45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8` which is
+  the private key for account
+  `a94f5374fce5edbc8e2a8697c15331677e6ebf0b`
+- to - called account, if the account is an Internally Owned Account it will
+  execute the wasm code.
+- value - value sent to the account.
+
+
+example:
+
+```yaml
+transaction:
+  data:
+  - '0x'
+  gasLimit:
+  - '0x6acfc0'
+  gasPrice: '0x01'
+  nonce: '0x04'
+  secretKey: "45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8"
+  to: 'deadbeef00000000000000000000000000000000'
+  value:
+    - '0'
+```
+
+## Filling and executing the tests
+
+Filling the test is the process to take the source Filler files and generate
+json files. The final json file contains the result of the execution as a
+post state hash.
+
+In order to "fill the tests" we need cpp-ethereum/testeth built with hera.
+
+Testeth also needs `lllc` (LLL Compiler) in order to fill tests cases (even if
+we are not using LLL code in our tests), `lllc` compiler is distributed along
+with [solidity][13], you may need to compile solidity/lllc and add it to your `$PATH`.
+
+### building cpp-ethereum with hera
+
+For building cpp ethereum you can use the commit specified in the [hera circleci file][12]
+
+For example, current commit used in hera circle ci is
+`b387b5713738da12444af8d3a0f6390f2c87b76e`, in order to build cpp-ethereum with
+hera we follow the following steps:
+
+- clone cpp-ethereum
+
+```
+git clone https://github.com/ethereum/cpp-ethereum.git
+```
+
+- go to the specific commit
+
+```
+cd cpp-ethereum/
+git reset --hard b387b5713738da12444af8d3a0f6390f2c87b76e
+```
+
+- update submodules
+
+```
+git submodule update --init
+```
+
+- update hera to the latest version (or the specific hera version you want to
```

Comment:
This is confusing. Updating hera like this shouldn't be necessary if the user has already run `git submodule update --init` (and maybe with `--recursive` as well--actually I'm not sure this is necessary for hera). Can we just keep the right version of hera pinned at all times? Maybe the note should say, "If you want to test against a specific version of Hera, you can do it like this..."